p2p/peer.go

```diff
@@ -34,6 +37,9 @@ type Peer interface {
 	Status() tmconn.ConnectionStatus
 	SocketAddr() *NetAddress // actual address of the socket
 
+	NewSend(Envelope) bool
+	NewTrySend(Envelope) bool
```

Comment:
Could we rather please name these something like `SendEnvelope` and `TrySendEnvelope`? It's closer to their actual function.

My understanding is that we'd deprecate `Send` and `TrySend` in v0.37, and then remove those methods in v0.38, leaving `SendEnvelope` and `TrySendEnvelope` in v0.38. Does that match with your understanding @williambanfield?

It would also be helpful to have potential deprecation notices on the interface's methods here too.

Reply:
> Could we rather please name these something like SendEnvelope and TrySendEnvelope? It's closer to their actual function.

Sure thing, done.

> My understanding is that we'd deprecate Send and TrySend in v0.37, and then remove those methods in v0.38, leaving SendEnvelope and TrySendEnvelope in v0.38. 

This is my inclination as well. The PR that this is a backport of does not include the old Send/Receive/Broadcast methods that operated on bytes at all.

> would also be helpful to have potential deprecation notices on the interface's methods here too.

done.